pkg/apis/core/validation/validation.go

```diff
@@ -7271,6 +7288,43 @@ func validateMatchLabelKeys(fldPath *field.Path, matchLabelKeys []string, labelS
 	return allErrs
 }
 
+// validateLabelKeysWithSelector tests that the label keys are a valid label name and are not already included in labelSelector.
+// It's intended to be used for matchLabelKeys or mismatchLabelKeys.
+func validateLabelKeysWithSelector(fldPath *field.Path, fldName string, labelKeys []string, labelSelector *metav1.LabelSelector) field.ErrorList {
+	if len(labelKeys) == 0 {
+		return nil
+	}
+
+	if labelSelector == nil {
+		return field.ErrorList{field.Forbidden(fldPath, "must not be specified when labelSelector is not set")}
+	}
+
+	var allErrs field.ErrorList
+	// labelKeysMap is keyed by label key and valued by the index of label key in labelKeys.
+	labelKeysMap := map[string]int{}
+	for i, key := range labelKeys {
+		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(key, fldPath.Index(i))...)
+		labelKeysMap[key] = i
+	}
+	labelSelectorKeys := sets.New[string]()
+	for key := range labelSelector.MatchLabels {
+		labelSelectorKeys.Insert(key)
+	}
+	for _, matchExpression := range labelSelector.MatchExpressions {
+		key := matchExpression.Key
+		if i, ok := labelKeysMap[key]; ok && labelSelectorKeys.Has(key) {
+			// Before validateLabelKeysWithSelector is called, the labelSelector has already got the selector created from matchLabelKeys/mismatchLabelKeys.
+			// Here, we found the duplicate key in labelSelector and the key is specified in labelKeys.
+			// Meaning that the same key is specified in both labelSelector and matchLabelKeys/mismatchLabelKeys.
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), key, fmt.Sprintf("exists in both %s and labelSelector", fldName)))
```

Comment:
But, wait. I just noticed one thing. Your question is on point if it comes to mismatchLabelKeys. mismatchLabelKeys will be notIn, so users may want to have a further labelSelector with the same key in addition to mismatchLabelKeys so that it can filter out more Pods to match. Let me modify the validation logic.